app/demo-loop/LDemoIO.hh

```diff
@@ -24,20 +24,17 @@ struct LDemoArgs
     // Problem definition
     std::string geometry_filename; //!< Path to GDML file
     std::string physics_filename;  //!< Path to ROOT exported Geant4 data
-
-    // Source definition (TODO: specify particle type, distribution, ...?)
-    double energy{}; //!< use hepmc3 input?
+    std::string hepmc3_filename;   //!< Path to Hepmc3 event data
 
     // Control
     unsigned int seed{};
-    size_type    num_tracks{};
```

Comment:
Maybe I should have named this as `max_num_tracks` -- the total limit for tracks in flight, aka the peak number of threads. The number of tracks is going to be different from the number of events/primaries. In theory the max could be greater than *or* less than the number of primaries in the event file.

Reply:
Yeah, I discussed that with @amandalund for a bit, and I was not sure if it would be better to have a hardcoded one or one based on the number of primaries (that we could calculate from the loaded hepmc3 primary data).

I can revert that back to the manual input from the json file.

Reply:
Ah gotcha. No, I think (at the beginning at least) we're going to need some user-configurable options for how to allocate memory: number of primaries, number of track initializers, number of events, etc.

Reply:
Okay. Reverting that...